fastcrypto/src/polynomial.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Most of the code below is based on code from https://github.com/celo-org/celo-threshold-bls-rs,
+// modified for our needs.
+//
+
+use crate::error::FastCryptoError;
+use crate::groups::{GroupElement, Scalar};
+use crate::traits::AllowedRng;
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use std::num::NonZeroU32;
+
+//// Types
+
+/// Indexes of shares (0 is reserved for the secret itself).
+pub type ShareIndex = NonZeroU32;
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct IndexedValue<A> {
+    pub index: ShareIndex,
+    pub value: A,
+}
+
+pub type Eval<A> = IndexedValue<A>;
+
+/// A polynomial that is using a scalar for the variable x and a generic
+/// element for the coefficients.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Poly<C>(Vec<C>);
+
+pub type PrivatePoly<C> = Poly<<C as GroupElement>::ScalarType>;
+pub type PublicPoly<C> = Poly<C>;
+
+//// Vector related operations.
+
+impl<C> Poly<C> {
```

Comment:
Ooc, is the celo paradigm the most popular + fast lib for polynomials?

Reply:
Not sure there is a "paradigm" as we only compute Lagrange coefficients and multiply them with group elements, it's the straightforward method (which many others use as well).
There are faster ways for some use cases, e.g., using FFTs for large values of t, but for our needs it should suffice for now.